preprocessing/preprocessing_one_hot_encoding.py

```diff
@@ -0,0 +1,119 @@
+import pandas as pd
+from pandas import DataFrame
+from typing import Tuple
+
+from sklearn.preprocessing import OneHotEncoder
+
+from preprocessing.preprocessing import get_preprocessed_brfss_dataset, diabetes_columns, oversample_dataset, \
+    undersample_dataset, get_train_test_split
+
+# Excludes yes/no columns
+ordinal_columns = ["GenHealth", "Checkup", "HighBP", "Income", "Age", "BMI", "Education", "Smoking", "PhysActivity"]
+
+# Includes yes/no columns
+all_ordinal_columns = ["GenHealth", "Healthcare", "MedCost", "Checkup", "HighBP", "HighChol", "HeartAttack", "AngiCoro",
+                       "Stroke", "Asthma", "Arthritis", "Kidney", "Sex", "Income", "SodiumSalt", "Age", "BMI",
+                       "Education", "Alcohol", "Smoking", "FruitCons", "VegetCons", "PhysActivity", "Muscles"]
+
+
+# Returns preprocessed dataset where all columns with ordinal values that are not simply yes/no are one hot encoded.
+# If the parameter is set to true, the target column will also be one hot encoded.
+# This function does not include sampling.
+def get_preprocessed_brfss_dataset_one_hot_encoded(target_one_hot_encoded=False) -> Tuple[DataFrame, DataFrame]:
+    dataset, target = get_preprocessed_brfss_dataset()
+    one_hot_encoder = OneHotEncoder()
+    encoded = pd.DataFrame(
+        one_hot_encoder.fit_transform(dataset[ordinal_columns]).toarray(),
+        columns=one_hot_encoder.get_feature_names_out())
+    dataset = dataset.join(encoded)
+    dataset = dataset.drop(columns=ordinal_columns)
+    if target_one_hot_encoded:
+        target = pd.DataFrame(one_hot_encoder.fit_transform(target).toarray(), columns=diabetes_columns)
+    return dataset, target
+
+
+# Returns a train/test split of the preprocessed dataset
+# where all columns with ordinal values that are not simply yes/no are one hot encoded.
+# If the parameter is set to true, the target column will also be one hot encoded.
+# This function does not include sampling.
+def get_preprocessed_brfss_dataset_one_hot_encoded_train_test_split(target_one_hot_encoded=False) \
+        -> Tuple[DataFrame, DataFrame, DataFrame, DataFrame]:
+    dataset, target = get_preprocessed_brfss_dataset_one_hot_encoded(target_one_hot_encoded)
+    data_train, data_test, target_train, target_test = get_train_test_split(dataset, target)
+    return data_train, data_test, target_train, target_test
```

Comment:
There are a few methods like this where the objects can be returned directly. That would make it more readable. The return type is clear anyway based on the type hints.
`return get_train_test_split(dataset, target)`

Reply:
Done